otherlibs/stdune/src/ansi_color.ml

```diff
@@ -285,6 +401,50 @@ let index_from_any str start chars =
   in
   go start
 
+let rec parse_styles (accu : Style.t list) l =
+  (* This function takes a list of strings, taken from splitting an Ansi code on
+     ';', and adds the parsed styles to the already accumulated styles. There is
+     some non-trivial interaction with parsing here. For example, 8-bit and
+     24-bit color codes need some lookahead and default colours need to be able
+     to override other styles. *)
+  match l with
+  | [] -> accu (* Parsing 8-bit foreground colors *)
+  | "38" :: "5" :: s :: l -> (
+    match Int.of_string s with
+    | None -> parse_styles accu l
+    | Some code -> `Fg_8_bit_color (RGB8.of_int code) :: parse_styles accu l)
```

Comment:
1. Why is this function no longer tail recursive?
2. Why are we accumulating the style in the beginning here? Shouldn't it be cons'd onto accu?